staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -139,38 +140,49 @@ type Requirement struct {
 //     of characters. See validateLabelKey for more details.
 //
 // The empty string is a valid value in the input values set.
-func NewRequirement(key string, op selection.Operator, vals []string) (*Requirement, error) {
-	if err := validateLabelKey(key); err != nil {
+// Note: We must keep the exported function return type as error.
```

Comment:
Just say: `Returned error, if not nil, is guaranteed to be *field.Error`.

But, we should be aggregating all errors.

Reply:
I already tried to return all errors. But I have to aggregate the field.ErrorList to ensure it's an `error`, then it can't be used in [here](https://github.com/kubernetes/kubernetes/pull/97538/files/0a3c5ca60cd3e38ae1b6c3093f415bd9882222db#diff-585e377f5b659eccad32e5d662e92f83e68a4b894e78014374affcaf1180e574R228), where requires a field.Erorr.